src/controllers/player-profile/find-all-player-profile-controller.ts

```diff
@@ -0,0 +1,17 @@
+import { type FindAllPlayerProfile } from '@/contracts/player-profile/find-all-player-profile'
+import { ok, serverError } from '@/helpers/http/http-helpers'
+import { type HttpRequest, type HttpResponse } from '@/types'
+import { type Controller } from '@nestjs/common/interfaces'
+
+export class FindAllPlayerProfileController implements Controller {
+  constructor (private readonly findAllPlayerProfile: FindAllPlayerProfile) {}
+
+  async handle (httpRequest: HttpRequest): Promise<HttpResponse> {
+    try {
+      const result = await this.findAllPlayerProfile.perform()
+      return ok(result)
+    } catch (error: any) {
```

Comment:
Aqui ao invés de any poderia usar o unknown. Usar o any desabilita todas as verificações de tipo. Qualquer valor pode ser atribuído a uma variável any, e essa variável pode ser usada de qualquer maneira, sem verificação de tipo. Já o unknown força você a realizar verificações de tipo antes de usar a variável. Isso garante que você saiba exatamente o que está lidando antes de manipular o valor.

Ao usar unknown, é garantido que, se precisar inspecionar ou manipular error, isso será feito de maneira segura, verificando o tipo antes de acessar as propriedades ou métodos.

Reply:
Atualizado tipagem